R/users.R

```diff
@@ -179,3 +179,23 @@ users_exist <- function(vault, users) {
 on_failure(users_exist) <- function(call, env) {
   paste0("Secret ", deparse(call$users), " do not exist")
 }
+
+#' @importFrom openssl fingerprint read_pubkey
+
+lookup_user <- function(key, vault = NULL) {
+  user <- NULL
+  vault <- find_vault(vault)
+  if (is.character(key)) {
+    key <- tryCatch(read_key(key),
+                    error = function(e) NULL)
+    if (is.null(key)) return(NULL)
+  }
+  fp <- fingerprint(key)
+  for (pubkey in dir(file.path(vault, "users"), pattern='\\.pem$')) {
+    pubkeyfile = file.path(vault, 'users', pubkey)
+    if (as.character(fp) == as.character(fingerprint(read_pubkey(pubkeyfile)))) {
+      user <- sub("\\.pem$", "", pubkey)
+    }
```

Comment:
You can return from the loop, no need to continue the loop if you already found the user.